stl/inc/ranges

```diff
@@ -728,6 +735,441 @@ namespace ranges {
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                ++_Current;
+                return *this;
+            }
+
+            constexpr decltype(auto) operator++(int) {
+                if constexpr (forward_range<_Base>) {
+                    auto _Tmp = *this;
+                    ++*this;
+                    return _Tmp;
+                } else {
+                    ++*this;
+                }
+            }
+
+            constexpr _Iterator& operator--() requires bidirectional_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _STL_VERIFY(_Parent != nullptr, "Cannot decrement value-initialized transform_view iterator");
+                if constexpr (forward_range<_Vw>) {
+                    _STL_VERIFY(_Current != _RANGES begin(_Parent->_Range),
+                        "Cannot decrement transform_view iterator before begin");
+                }
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                --_Current;
+                return *this;
+            }
+            constexpr _Iterator operator--(int) requires bidirectional_range<_Base> {
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            constexpr void _Verify_offset(const difference_type _Off) const noexcept
+                requires random_access_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL == 0
+                (void) _Off;
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 / _ITERATOR_DEBUG_LEVEL != 0 vvv
+                _STL_VERIFY(_Off == 0 || _Parent, "cannot seek value-initialized transform_view iterator");
+
+                if constexpr (_Offset_verifiable_v<iterator_t<_Base>>) {
+                    _Current._Verify_offset(_Off);
+                } else {
+                    if (_Off < 0) {
+                        if constexpr (sized_sentinel_for<iterator_t<_Base>, iterator_t<_Base>>) {
+                            _STL_VERIFY(_Off >= _RANGES begin(_Parent->_Range) - _Current,
+                                "cannot seek transform_view iterator before begin");
+                        }
+                    } else if (_Off > 0) {
+                        if constexpr (sized_sentinel_for<sentinel_t<_Base>, iterator_t<_Base>>) {
+                            _STL_VERIFY(_Off <= _RANGES end(_Parent->_Range) - _Current,
+                                "cannot seek transform_view iterator after end");
+                        } else if constexpr (sized_sentinel_for<iterator_t<_Base>,
+                                                 iterator_t<_Base>> && sized_range<_Base>) {
+                            const auto _Size = _RANGES distance(_Parent->_Range);
+                            _STL_VERIFY(_Off <= _Size - (_Current - _RANGES begin(_Parent->_Range)),
+                                "cannot seek transform_view iterator after end");
+                        }
+                    }
+                }

```

Comment:
I feel like this could be a usefull helper function if passed a error message

Reply:
I'd like to put off refactoring this for future work. I think there will be more places we can use it, but I hate to make a separate function before we need it. Doubly so given that we kind of need to merge this tomorrow.